src/neuroglancer/widget/segment_set_widget.ts

```diff
@@ -69,79 +83,204 @@ export class SegmentSetWidget extends RefCounted {
     this.updateClearButtonVisibility();
   }
 
+  private anyRootSegments = () => {
+    return this.displayState.rootSegments.size > 0;
+  }
+
+  private anyHiddenRootSegments = () => {
+    return this.displayState.hiddenRootSegments!.size > 0;
+  }
+
   private updateClearButtonVisibility() {
-    let {clearButton} = this;
-    clearButton.style.display = (this.displayState.rootSegments.size > 0) ? '' : 'none';
+    const {clearButton} = this;
+    clearButton.style.display = (this.anyRootSegments() || this.anyHiddenRootSegments()) ? '' : 'none';
   }
 
-  private handleSetChanged(x: Uint64|Uint64[]|null, added: boolean) {
-    this.updateClearButtonVisibility();
-    let {items} = this;
+  private clearItems() {
+    const {itemContainer, clearButton, enabledItems, disabledItems} = this;
+    while (true) {
+      const lastElement = itemContainer.lastElementChild!;
+      if (lastElement === clearButton) {
+        break;
+      }
+      itemContainer.removeChild(lastElement);
+    }
+    enabledItems.clear();
+    disabledItems.clear();
+  }
+
+  private handleEnabledSetChanged(x: Uint64|Uint64[]|null, added: boolean) {
```

Comment:
Missing the call to `this.updateClearButtonVisibility();` here to check if the `clear-button` is still required